frontend/src/employee-frontend/components/income-statements/IncomeStatementsPage.tsx

```diff
@@ -156,29 +159,34 @@ export default React.memo(function IncomeStatementsPage() {
     incomeStatements: { searchFilters }
   } = useContext(InvoicingUiContext)
 
-  const [incomeStatements] = useApiState(() => {
-    const { sentStartDate, sentEndDate } = searchFilters
-    if (sentStartDate && sentEndDate && sentStartDate.isAfter(sentEndDate)) {
-      return Promise.resolve(Success.of({ data: [], pages: 0, total: 0 }))
-    }
-
-    return getIncomeStatementsAwaitingHandlerResult({
-      body: {
-        page,
-        pageSize,
-        sortBy,
-        sortDirection,
-        areas: searchFilters.area.length > 0 ? searchFilters.area : null,
-        providerTypes:
-          searchFilters.providerTypes.length > 0
-            ? searchFilters.providerTypes
-            : null,
-        sentStartDate: searchFilters.sentStartDate ?? null,
-        sentEndDate: searchFilters.sentEndDate ?? null,
-        placementValidDate: searchFilters.placementValidDate ?? null
-      }
-    })
-  }, [page, sortBy, sortDirection, searchFilters])
+  const incomeStatements = useQueryResult(
```

Comment:
nice, I think I can use this as an example to solve most open issues in my similar invoice search refactor